src/views/onboarding/Onboarding.tsx

```diff
@@ -187,6 +305,15 @@ function OnboardingComponent({ productId }: { productId: string }) {
         request_id: requestIdRef.current,
         product_id: productId,
       });
+      setOutcome(outcomeContent[outcome]);
```

Comment:
it's better to invoke setOutcome just once: setOutcome is a setter of useState and each invocation of it will schedulate a batch processing to evaluate the changes on the DOM, after that all the scheduled batch are completed, the DOM changes are applied, so finaly the final user will not see any intermediate result.
But reduce the number of this invocation (when possible) will reduce this evaluation.


So, call the setOutcome into the else branch of the following else